ADR/0047-movable-build-tests.md

```diff
@@ -0,0 +1,34 @@
+# 47. Attestable Build-Time Tests in Integration Service
+
+Date: 2025-04-06
+
+## Status
+
+Proposed
+
+## Context
+
+Some of the security scans run during the build pipeline take a long time to
+run. This forces users to wait longer to be notified of the results of their
+builds.  In order to receive more immediate feedback, users have requested that
+they be able to run security scans during their integration tests.
```

Comment:
While not necessarily incorrect, I feel like we can be more clear here.

```suggestion
Konflux makes it easy to run many security checks immediately after builds. While
the tests are shifted left (good), the checks are serialized to run immediately after the
builds which increase the feedback time (bad). Konflux has the ability to trigger additional
PipelineRuns immediately after the completion of artifact builds via IntegrationTestScenarios.

This architecture additionally improves the user experience as it allows these checks to be rerun
without having to rebuild the artifacts themselves, further decreasing feedback time upon failed
post-build TaskRuns.
```

Reply:
I think Ryan's original is more clear. It states the context that brings us to the decision, without yet stating the decision or its consequences.

Reply:
I agree after re-reading. I don't like what I proposed anymore. :)